sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -1343,17 +1343,41 @@ class HiveDDLSuite
       sql("INSERT INTO t SELECT 2, 'b'")
       checkAnswer(spark.table("t"), Row(9, "x") :: Row(2, "b") :: Nil)
 
-      val e = intercept[AnalysisException] {
-        Seq(1 -> "a").toDF("i", "j").write.format("hive").partitionBy("i").saveAsTable("t2")
-      }
-      assert(e.message.contains("A Create Table As Select (CTAS) statement is not allowed " +
-        "to create a partitioned table using Hive"))
-
       val e2 = intercept[AnalysisException] {
         Seq(1 -> "a").toDF("i", "j").write.format("hive").bucketBy(4, "i").saveAsTable("t2")
       }
       assert(e2.message.contains("Creating bucketed Hive serde table is not supported yet"))
 
+      try {
+        spark.sql("set hive.exec.dynamic.partition.mode=nonstrict")
```

Comment:
I think we can use `withSQLConf` instead of `try .. finally ..`.

```scala
withSQLConf("hive.exec.dynamic.partition.mode" -> "nonstrict") {
...
}
```